### PR TITLE
Feature/mztabm

### PR DESCRIPTION
**Pull request by sneumann**
This PR adds 
1) a possibly breaking change where the special characters like space, [ and ] are *not* modified upon readin, see discussion in #501 
2) A write.mzTab method which writes an MzTab object back to disc.

Points for discussion:
1) Is there any objection to the `check.names=FALSE` change ? 
2) is the implementation of `write.mzTab()` as method OK ? 
3) Which "flavour" of mzTab do we want to write w.r.t. to number of columns in the different sections ? Should it be rectangular ? I think there are cases where we need to "trim" empty columns before creating the MzTab slots.

Yours, Steffen

### CODE REVIEW

**Comment by lgatto** [comment]
Could you please send a PR per requested change. 

- The first one would be the change to `check.names = FALSE`, which I am happy with. With that PR in, it will be possible to check that there isn't anything that breaks. 
- As for `write.MzTab`, it should be `writeMzTab` and a simple function will do. 
- We try to adhere to [coding style](https://rformassspectrometry.github.io/RforMassSpectrometry/articles/RforMassSpectrometry.html#coding-style), which I am happy to help fix. But that will only be easy if you provide specific PRs (see first point above).

File: tests/testthat/test_MzTab.R [hunk 1]
```diff
@@ -11,7 +11,7 @@ test_that("MzTab creation and accessors", {
     expect_null(show(xx))
     ## Accessors
     expect_is(metadata(xx), "list")
-    expect_identical(length(metadata(xx)), 64L)
+    expect_identical(length(metadata(xx)), 65L)
```
Comment (sneumann) [inline tests/testthat/test_MzTab.R:14]: Turns out that the previous implementation without `headers=FALSE` swallowed the first line of the MTD section. After fixing that there is one more metadata line.

**Comment by lgatto** [comment]
Are there more PRs to come, or should I push to Bioconductor now?

**Comment by sneumann** [comment]
There are no imminent PRs. What's left is a decision about the dimensionality of things. 
Some mzTab files have columns of empty tabs to the right of the actual data.frame. 
R happily imagines some names like "V8" "V9" for such columns. This needs to be discussed 
with mzTab folks, possibly me reading specs again. I would opt for some (optional ?)
trimming of the MzTab@SmallMolecules data.frame upon reading. 
But as said, that's not coming soon. Yours, Steffen

**Comment by lgatto** [comment]
Pushed to Bioc.

**Pull request merged**